ring/util.go

```diff
@@ -193,16 +192,13 @@ func getFirstAddressOf(names []string, logger log.Logger) (string, error) {
 		if ip := filterIPs(addrs); ip != "" {
 			ipAddr = ip
 		}
-		if strings.HasPrefix(ipAddr, `169.254.`) || ipAddr == "" {
```

Comment:
Right. Then the `filterIPs()` comment is wrong now, because it says "// filterIPs attempts to return the first non automatic private IP (APIPA / 169.254.x.x) if possible, only returning APIPA if available and no other valid IP is found." but you actually removed the line `ipAddr = v.IP.String()` so the APIPA can never be returned. This is basically another regression. I would recommend to keep this PR tight to just adding IPv6 support only, without introducing regressions in the logic.

Reply:
I've made an update here to use `net.IP` until the final return so that we can make use of the helper methods.  I believe I avoided the regression now.